File: plugins/ls-scripting/skills/bun-script/assets/template-complex.ts
#!/usr/bin/env bun

import { cli } from "cleye";

const argv = cli({
  name: "script-name",
  help: {
    description: "Description of what the script does.",
  },
  flags: {
    first: {
      type: String,
      description: "Description of the first flag.",
    },
    second: {
      type: String,
      description: "Description of the second flag.",
    },
  },
});

// Check for invalid options
const invalidOption = Object.keys(argv.unknownFlags)[0];

if (invalidOption) {
  console.error(`Error: The option --${invalidOption} is invalid.`);
  argv.showHelp();
  process.exit(1);
}

// Validate required arguments
if (!argv.flags.first) {
  console.error("Error: The --first flag is required.");
  argv.showHelp();
  process.exit(1);
}

if (!argv.flags.second) {
  console.error("Error: The --second flag is required.");
  argv.showHelp();
  process.exit(1);
}

// TODO: Implement script

File: plugins/ls-scripting/skills/ruby-script/assets/template-complex.rb
#!/usr/bin/env ruby
# frozen_string_literal: true

require "optparse"

def print_help
  puts <<~HELP
    Usage: script-name [options]

    Description of what the script does.

    Options:

      --first <value>     Description of the first flag.
      --second <value>    Description of the second flag.
      --help              Show this help message and exit.
  HELP
end

first = nil
second = nil

parser = OptionParser.new do |opts|
  opts.on("--first VALUE") { first = _1 }
  opts.on("--second VALUE") { second = _1 }
  opts.on("--help") do
    print_help
    exit 0
  end
end

begin
  parser.parse!
rescue OptionParser::InvalidOption => e
  warn "Error: The option #{e.args.first} is invalid."
  warn
  print_help
  exit 1
end

# Validate required arguments
if first.nil? || first.empty?
  warn "Error: The --first flag is required."
  warn
  print_help
  exit 1
end

if second.nil? || second.empty?
  warn "Error: The --second flag is required."
  warn
  print_help
  exit 1
end

# TODO: Implement script
